DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UnsupportedViewRecorder.swift

```diff
@@ -10,17 +10,19 @@ import SwiftUI
 
 internal struct UnsupportedViewRecorder: NodeRecorder {
     // swiftlint:disable opening_brace
-    private let unsupportedViewsPredicates: [(UIView) -> Bool] = [
-        { $0 is UIProgressView },
-        { $0 is UIWebView },
-        { $0 is WKWebView },
-        { $0 is UIActivityIndicatorView },
-        { String(reflecting: type(of: $0)).contains("SwiftUI") }
+    private let unsupportedViewsPredicates: [(UIView, ViewTreeRecordingContext) -> Bool] = [
+        { _, context in context.viewControllerContext.isRootView(of: .safari) },
+        { _, context in context.viewControllerContext.isRootView(of: .activity) },
+        { view, _ in view is UIProgressView },
+        { view, _ in view is UIWebView },
+        { view, _ in view is WKWebView },
+        { view, _ in view is UIActivityIndicatorView },
+        { view, _ in String(reflecting: type(of: view)).contains("SwiftUI") }
```

Comment:
suggestion/ Looks like we could get more clarity and coherence if this is determined equally to `.activity` or `.safari`. Now "SwiftUI content" is recognised as an agreement and logic in multiple code units, but we could do it all in one place:
```swift
{ _, context in context.viewControllerContext.isRootView(of: .swiftUI) },
```